frontend_csv/src/parser/ast_msg.rs

```diff
@@ -32,38 +33,38 @@ pub struct AstMsg {
     pub msg_val: String,
 }
 
-pub struct InitMsg {
+pub struct Msg {
     pub raw: Vec<(AstMsgLineType, Vec<String>)>,
     pub msg_type: AstMsgType,
-    pub msg_val: String,
+    pub msg_val: Vec<Vec<String>>,
     pub msg_name: String,
     pub tlvs_name: Vec<String>,
 }
 
 pub trait AstMsgTrait {
     fn get_msg_type(&self) -> AstMsgType;
-    fn get_values(&self) -> String;
+    fn get_values(&self) -> Vec<Vec<String>>;
     fn check_valid(&mut self) -> bool;
 }
 
-impl InitMsg {
-    pub fn new(lines: Vec<(AstMsgLineType, Vec<String>)>) -> InitMsg {
-        return InitMsg {
-            raw: lines,
-            msg_type: AstMsgType::Init,
-            msg_val: String::new(),
+impl Msg {
+    pub fn new(lines: Vec<(AstMsgLineType, Vec<String>)>) -> Msg {
+        return Msg {
+            raw: lines.clone(),
+            msg_type: AstMsgType::None,
+            msg_val: lines.into_iter().map(|p| p.1).collect(),
             msg_name: String::new(),
             tlvs_name: Vec::new(),
         };
     }
 }
 
-impl AstMsgTrait for InitMsg {
+impl AstMsgTrait for Msg {
     fn get_msg_type(&self) -> AstMsgType {
         return self.msg_type.clone();
     }
 
-    fn get_values(&self) -> String {
+    fn get_values(&self) -> Vec<Vec<String>> {
         return self.msg_val.clone();
     }
```

Comment:
```suggestion
    fn get_values(&self) -> Vec<Vec<String>> {
        return self.msg_val.clone();
    }
```

Why do you node access the directory to `self.msg_type`? what is the meaning of this? force rust to make a new copy?